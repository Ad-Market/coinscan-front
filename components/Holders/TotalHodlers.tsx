import { ApolloError } from '@apollo/client';
// import { TotalHoldersQuery } from '../../generated/graphql';
// import { gql, useQuery } from '@apollo/client';
import { DailyHodlersStatesQuery } from '../../generated/graphql';
import { Loading } from '../Wallets/Wallets.styled';
import Growth from '../Growth';

// const GET_TOTAL_HOLDERS = gql`
//   query TotalHolders {
//     holdersCounters {
//       id
//       count
//     }
//   }
// `;

type TotalHoldersProps = {
  loading?: boolean;
  error?: ApolloError;
  data?: DailyHodlersStatesQuery;
};

function getPercentChange(
  current: number | undefined,
  prev: number | undefined
): number | undefined {
  if (current && prev) {
    const percent = (current / prev) * 100 - 100;
    return percent;
  }

  return;
}

const TotalHodlers = ({ loading, error, data }: TotalHoldersProps) => {
  // const { loading, error, data } =
  //   useQuery<TotalHoldersQuery>(GET_TOTAL_HOLDERS);

  if (loading) return <Loading>Loading...</Loading>;
  if (error) return <div>{error.toString()}</div>;

  const currentHoldersCount = data?.dailyHoldersStates[0].count;
  const oneDayHoldersCount = data?.dailyHoldersStates[1].count;
  const sevenDaysHoldersCount = data?.dailyHoldersStates[6].count;
  const thirtyDaysHoldersCount = data?.dailyHoldersStates[29].count;

  return (
    <Growth
      value={currentHoldersCount}
      oneday={getPercentChange(currentHoldersCount, oneDayHoldersCount)}
      sevendays={getPercentChange(currentHoldersCount, sevenDaysHoldersCount)}
      thirtydays={getPercentChange(currentHoldersCount, thirtyDaysHoldersCount)}
    />
  );
};

export default TotalHodlers;
