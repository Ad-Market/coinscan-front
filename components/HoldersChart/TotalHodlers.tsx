import { useState } from 'react';
import { TotalHoldersQuery } from '../../generated/graphql';
import { gql, useQuery } from '@apollo/client';
import { Loading } from '../Wallets/Wallets.styled';

const GET_TOTAL_HOLDERS = gql`
  query TotalHolders {
    holdersCounters {
      id
      count
    }
  }
`;

const TotalHodlers = () => {
  const { loading, error, data, fetchMore } =
    useQuery<TotalHoldersQuery>(GET_TOTAL_HOLDERS);

  if (loading) return <Loading>Loading...</Loading>;
  if (error) return <div>{error.toString()}</div>;

  return (
    <div>
      <span>Total holders: </span>
      {data?.holdersCounters[0]?.count}
    </div>
  );
};

export default TotalHodlers;
