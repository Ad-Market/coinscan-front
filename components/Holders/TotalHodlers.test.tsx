import { screen, render } from '@testing-library/react';
import TotalHodlers from './TotalHodlers';
import { testDataDailyHolders } from './test.fixture';

test('render total holders', async () => {
  const { container } = render(
    <TotalHodlers data={{ dailyHoldersStates: testDataDailyHolders }} />
  );
  await screen.findByText('6,341');
  await screen.findByText('0%');
  await screen.findByText('-0.14%');
  await screen.findByText('+0.89%');
  expect(container).toMatchSnapshot();
});
