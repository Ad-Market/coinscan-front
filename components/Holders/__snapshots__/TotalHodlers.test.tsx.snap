// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render total holders with loading pre-screen: loaded 1`] = `
"Snapshot Diff:
- First value
+ Second value

  <DocumentFragment>
    <div
-     class=\\"css-13xyho9\\"
+     class=\\"MuiBox-root css-0\\"
+   >
+     <div
+       class=\\"MuiBox-root css-1jolfkj\\"
+     >
+       Total Holders
+     </div>
+     <div
+       class=\\"MuiBox-root css-1n4i7xh\\"
+     >
+       6,341
+     </div>
+     <div
+       class=\\"MuiBox-root css-j7qwjs\\"
+     >
+       <div
+         class=\\"MuiBox-root css-1xhj18k\\"
+       >
+         <div
+           class=\\"MuiBox-root css-daxnmb\\"
+         >
+           0.00%
+         </div>
+         <div
+           class=\\"MuiBox-root css-1mvg5zt\\"
+         >
+           vs. last day
+         </div>
+       </div>
+       <div
+         class=\\"MuiBox-root css-1xhj18k\\"
+       >
+         <div
+           class=\\"MuiBox-root css-1d6tm0y\\"
+         >
+           -0.14%
+         </div>
+         <div
+           class=\\"MuiBox-root css-1mvg5zt\\"
+         >
+           vs. last 7 days
+         </div>
+       </div>
+       <div
+         class=\\"MuiBox-root css-1xhj18k\\"
        >
-     Loading...
+         <div
+           class=\\"MuiBox-root css-s7l08z\\"
+         >
+           +0.89%
+         </div>
+         <div
+           class=\\"MuiBox-root css-1mvg5zt\\"
+         >
+           vs. last 30 day
+         </div>
+       </div>
+     </div>
    </div>
  </DocumentFragment>"
`;

exports[`render total holders with loading pre-screen: loading 1`] = `
<DocumentFragment>
  <div
    class="css-13xyho9"
  >
    Loading...
  </div>
</DocumentFragment>
`;
